### ISSUE TITLE
Account type text selfdefine

### ISSUE THREAD

**Issue by renato-mmex** [moneymanagerex/moneymanagerex#3894]
# Description of the feature
**I would welcome the possibility of defining these texts myself. This would allow you to design various reports as you wish.**

![image](https://user-images.githubusercontent.com/76920138/135529074-5d8877ae-abaa-4f45-9639-a745bc7362fd.png)

**Comment by tactilis** [comment]
@renato-mmex
> This would allow you to design various reports as you wish

Could you provide more detail please about what you mean by "designing various reports as you wish"?

Do you mean simply that the name of the nodes in the Navigation tree and the section titles on the Home page would change to your new names - or do you mean something more than this?

Could you explain the use case and give some examples please of how you would like to change the names and where these new names would be seen in the reports.

**Comment by renato-mmex** [comment]
just by the way the selection "Checking" should actually be "Bank Accounts".

What is the use of empty columns in the reports, I would rather have the possibility to summarize the accounts I have with a bank.

Here is some more info about it:
<img width="395" alt="MMEX31" src="https://user-images.githubusercontent.com/76920138/135677839-f1a8b1a3-aa3f-4883-9f74-ab3e18908b29.png">
<img width="434" alt="MMEX32" src="https://user-images.githubusercontent.com/76920138/135677851-401e8241-4b01-4ec5-8814-a5581f0295ba.png">
<img width="429" alt="MMEX33" src="https://user-images.githubusercontent.com/76920138/135677860-9d61c9cc-829b-4c71-b781-750be40ca239.png">

**Comment by whalley** [comment]
So you want to be able to create arbitrary 'Account Groups' so that you can choose to put all accounts with a specific bank together rather than the fixed groups we have today. In you case that grouping is via institution. That grouping would be represented on the navigation tree and reports where the current grouping are used.
https://github.com/moneymanagerex/moneymanagerex/issues/1866

I don't think it is an unreasonable request and would align MMEX with the way some other finance software approaches this.

**Comment by renato-mmex** [comment]
MMEX is a "modern" software because it is built on a Sqlite database.
Unfortunately, there are elements that are not used in this way, which I can understand quite well due to the earlier programming possibilities.
If these remnants (fixed tables) could be eliminated, MMEX would be a bit more flexible from my point of view.

**Comment by tactilis** [comment]
We clearly need a long term strategy for database schema upgrades for the MMEX desktop app and the Android app.

At present, enhancements to MMEX have to be 'shoehorned' into the existing database schema so that compatibility with the Android app is not broken. This obviously limits what can be done in the desktop app.

I have little idea how to solve this (I don't know that state of Android development or even how widely the Android app is used), but without a roadmap to guide the future development of the database and deal with desktop/Android incompatibilities, then MMEX will remain constrained.

**Comment by whalley** [comment]
> We clearly need a long term strategy for database schema upgrades for the MMEX desktop app and the Android app.
> 
> At present, enhancements to MMEX have to be 'shoehorned' into the existing database schema so that compatibility with the Android app is not broken. This obviously limits what can be done in the desktop app.

Yes, as you say this is limiting our ability to evolve MMEX. Whilst I don't favour it we may have to leave users on the Android version behind on an older DB until the Android version can be updated. I'm not involved in the Android work so don't know what the capacity is to co-ordinate DB changes.  I can see this causing headaches as happened with the 1.4.x stream of releases.

**Comment by renato-mmex** [comment]
There are still big companies that have / "create" such problems, for example Microsoft with Windows 10 > Windows 11. According to my estimate, well over 60% remain on Windows 10 because the hardware is too old.  I look after about 23 PCs, only 5 meet the required conditions.

I doubt that there will be a volunteer in the near future who will regularly bring the Android version up to the level of the desktop version.

**Comment by vomikan** [comment]
> > We clearly need a long term strategy for database schema upgrades for the MMEX desktop app and the Android app.
> > At present, enhancements to MMEX have to be 'shoehorned' into the existing database schema so that compatibility with the Android app is not broken. This obviously limits what can be done in the desktop app.
> 
> Yes, as you say this is limiting our ability to evolve MMEX. Whilst I don't favour it we may have to leave users on the Android version behind on an older DB until the Android version can be updated. I'm not involved in the Android work so don't know what the capacity is to co-ordinate DB changes. I can see this causing headaches as happened with the 1.4.x stream of releases.

We have made a lot of fixes and improvements within the DB version 7. As a result, it turned out to be a good release.
I think it's worth announcing that we will move on without looking back at android. I suggest switching to version 1.6.x with the changes that are required.

**Comment by renato-mmex** [comment]
From the time when the database is no longer compatible with the Android version, you could point it out during the installation of MMEX. 
Question: are there other MMEX versions that would be affected?
Are there any other additions that should/must be included in a new database version. ?  Which ?

**However, I would prefer this to take place in a MMEX version 2.0, so that it is clear that this is a new MMEX generation.**
